APIs/AuthorizationAPI.raml

```diff
@@ -116,13 +110,31 @@ types:
         description: Client-generated, high-entropy cryptographic random STRING with a minimum length of 43 characters and a maximum length of 128 characters
         type: string
         required: false
+  TokenCredentialsRequest:
+    properties:
+      grant_type:
+        description: The OAuth 2.0 grant type of the request
+        type: string
+        required: true
+      client_id:
+        description: The identifier of the client authenticating
+        type: string
+        required: true
+      client_assertion_type:
```

Comment:
Are these the correct keys to be using? In [RFC7521](https://tools.ietf.org/html/rfc7521#section-4.2) and [RFC7523](https://tools.ietf.org/html/rfc7523#section-2.2) it uses these keys for client _authentication_ but lower down we've said that these are the keys required in the body for a token authrorization request 

RFC7521 says "Token endpoints can differentiate between assertion-based credentials
   and other client credential types by looking for the presence of the
   "client_assertion" and "client_assertion_type" parameters, which will
   only be present when using assertions for client authentication."

In the examples for Authorization requests in [RFC7521 Section 4](https://tools.ietf.org/html/rfc7521#section-4.1) and [RFC 7523 Section 4](https://tools.ietf.org/html/rfc7523#section-4) it gives `grant_type` and `assertion` as the keys in the body of the request

Reply:
The way how I read the RFCs, there are 2 ways of using the JWT.

1. use for Authorization Grants, doing so the token request will only require the `grant_type` and `assertion`.

2. use for Client Authentication, that require the `client_assertion_type`, `client_assertion` and `client_id`.

For our purpose we are interested using it for case no.2, the Client Authentication.

Another point I would like to make, once client is registered with `"token_endpoint_auth_method": "private_key_jwt"`, client will always require to provide the client_assertion_type, client_assertion (JWT) and client_id while interact with the token endpoint no matter client_credentials or authorization_code grant_type is used. Do you think we need to add that information in our grant examples?

Reply:
Thanks for replying to my query Simon. Things were cleared up in [Section 6.2 of RFC7521](https://tools.ietf.org/html/rfc7521#section-6.2) where it explicitly states that the _authentication_ method should be used when the client credentials grant is being used. Perhaps an example of each of the forms that the client credentials grant can take might make things clearer